### PR TITLE
Enable OccupancyGridNode and remove generating of assets in the node.

### LINKED ISSUE #413: node crashes if any node subscribes to OccupancyGrid
It crashes in [map_builder_bridge.cc#170](https://github.com/googlecartographer/cartographer_ros/blob/master/cartographer_ros/cartographer_ros/map_builder_bridge.cc#L170). 

* rviz was subscribing to occupancy grid topic.
* trajectory_id 0 was for loaded map from serialized data.

```
F0705 19:26:54.898905 26894 map_builder_bridge.cc:170] Check failed: trajectory_options_.count(0) == 1 (0 vs. 1) 
[/cartographer_node FATAL 1499275614.899195430, 1498225574.110075237]: F0705 19:26:54.000000 26894 map_builder_bridge.cc:170] Check failed: trajectory_options_.count(0) == 1 (0 vs. 1) 
*** Check failure stack trace: ***575.107289   Duration: 1.908086 / 591.608141               
    @     0x7f9bec9f95cd  google::LogMessage::Fail()
    @     0x7f9bec9fb433  google::LogMessage::SendToLog()
    @     0x7f9bec9f915b  google::LogMessage::Flush()
    @     0x7f9bec9fbe1e  google::LogMessageFatal::~LogMessageFatal()
    @           0x51fe83  cartographer_ros::MapBuilderBridge::BuildOccupancyGrid()
    @           0x4fa7ee  cartographer_ros::Node::SpinOccupancyGridThreadForever()
    @     0x7f9be9064c80  (unknown)
    @     0x7f9bec5b76ba  start_thread
    @     0x7f9be8ad33dd  clone
    @              (nil)  (unknown)
```

### PR DESCRIPTION
**Pull request by SirVer**
This will allow us to keep way less information in memory for SLAMing and fixes crashes related to multi trajectory (e.g. this fixes #413).

It also means that to get to an X-Ray or a map, users need to run the asset writer now after SLAMing, which is inconvenient.

### CODE REVIEW

File: cartographer_ros/launch/offline_backpack_2d.launch [hunk 1]
```diff
@@ -26,7 +26,10 @@
           -configuration_basename backpack_2d.lua
           -urdf_filename $(find cartographer_ros)/urdf/backpack_2d.urdf
           -bag_filenames $(arg bag_filenames)"
-      output="screen">
```
Comment (wohe) [inline cartographer_ros/launch/offline_backpack_2d.launch]: Revert?

File: cartographer_ros/launch/offline_backpack_3d.launch [hunk 2]
```diff
@@ -25,9 +25,11 @@
           -configuration_directory $(find cartographer_ros)/configuration_files
           -configuration_basename backpack_3d.lua
           -urdf_filename $(find cartographer_ros)/urdf/backpack_3d.urdf
-          -bag_filenames $(arg bag_filenames)"
-      output="screen">
```
Comment (wohe) [inline cartographer_ros/launch/offline_backpack_3d.launch]: Revert?

File: docs/source/ros_api.rst [hunk 3]
```diff
@@ -100,9 +100,10 @@ finish_trajectory (`cartographer_ros_msgs/FinishTrajectory`_)
   Finishes the given `trajectory_id`'s trajectory by running a final optimization.
 
 write_assets (`cartographer_ros_msgs/WriteAssets`_)
```
Comment (wohe) [inline docs/source/ros_api.rst]: This should probably be renamed: `write_state`? Same with the ROS message.

File: docs/source/ros_api.rst [hunk 4]
```diff
@@ -100,9 +100,10 @@ finish_trajectory (`cartographer_ros_msgs/FinishTrajectory`_)
...
-  Writes artifacts (e.g. the map) to disk. The `stem` argument is used as a prefix
-  for the various files which are written. Files will usually end up in `~/.ros` or
-  `ROS_HOME` if it is set.
+  Writes the current internal state to disk. The `stem` argument is used as a prefix
```
Comment (wohe) [inline docs/source/ros_api.rst]: This only writes a pbstream file now, right? Change from `stem` to `filename`?

File: cartographer_ros/cartographer_ros/assets_writer.cc [hunk 5]
```diff
@@ -1,105 +0,0 @@
...
-#include "cartographer/common/make_unique.h"
-#include "cartographer/common/time.h"
-#include "cartographer/io/file_writer.h"
-#include "cartographer/io/null_points_processor.h"
-#include "cartographer/io/ply_writing_points_processor.h"
-#include "cartographer/io/points_processor.h"
-#include "cartographer/io/xray_points_processor.h"
-#include "cartographer/mapping/proto/trajectory.pb.h"
-#include "cartographer/mapping_2d/proto/range_data_inserter_options.pb.h"
-#include "cartographer_ros/map_writer.h"
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/assets_writer.cc]: I think this is the only use of `map_writer`. Remove as well?

Comment (dantwinkler) [inline cartographer_ros/cartographer_ros/assets_writer.cc]: What is the recommended way to generate the pgm/yaml from `write_state` (map-state.dat) and `assets_writer_main`.

Comment (SirVer) [inline cartographer_ros/cartographer_ros/assets_writer.cc]: 1. Start Cartographer (offline or online)
2. Start the Occupancy Grid node.
3. use `map_saver` from http://wiki.ros.org/map_server

**Pull request merged**